File: src-tauri/src/dto.rs
use serde::Serialize;

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct Artista {
    pub id: i32,
    pub nombre: String,
    pub pais: String,
    pub fechaNac: String,
    pub biografia: Option<String>,
    pub obras: Option<Vec<Obra>>, // Relación opcional con las obras
    pub fechaRegistro: String,
    pub fechaModificacion: String,
}

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct Obra {
    pub id: i32,
    pub titulo: String,
    pub tipo: String,
    pub precio: f64,
    pub descripcion: String,
    pub fechaCreacion: String,
    pub estado: String,
    pub idAutor: i32,
    pub detalles: Option<Vec<DetalleVenta>>, // Relación opcional con detalles de ventas
    pub fechaRegistro: String,
    pub fechaModificacion: String,
}

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct Cliente {
    pub id: i32,
    pub nombre: String,
    pub direccion: String,
    pub telefono: String,
    pub email: String,
    pub compras: Option<Vec<Venta>>, // Relación opcional con las ventas
    pub fechaRegistro: String,
    pub fechaModificacion: String,
}

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct Venta {
    pub id: i32,
    pub fecha: String,
    pub idCliente: i32,
    pub total: f64,
    pub detalles: Option<Vec<DetalleVenta>>, // Relación opcional con detalles de ventas
    pub fechaRegistro: String,
    pub fechaModificacion: String,
}

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct DetalleVenta {
    pub id: i32,
    pub idVenta: i32,
    pub idObra: i32,
}

#[allow(non_snake_case)]
#[derive(Serialize, Debug)] // Habilita la serialización para enviar como JSON
pub struct Empleado {
    pub id: i32,
    pub nombre: String,
    pub cargo: String,
    pub telefono: String,
    pub email: String,
    pub fechaContratacion: String,
    pub fechaRegistro: String,
    pub fechaModificacion: String,
}

File: src/backend/dto.ts

export interface Artista {
    id: number;
    nombre: string;
    pais: string;
    fechaNac: string;
    biografia?: string;
    obras?: Obra[]; // Un artista puede tener muchas obras
    fechaRegistro: string;
    fechaModificacion: string;
}

export interface Obra {
    id: number;
    titulo: string;
    tipo: string;
    precio: number;
    descripcion: string;
    fechaCreacion: string;
    estado: string;
    idAutor: number;
    autor?: Artista; // Una obra pertenece a un artista
    detalles?: DetalleVenta[]; // Una obra puede estar en muchas ventas
    fechaRegistro: string;
    fechaModificacion: string;
}

export interface Cliente {
    id: number;
    nombre: string;
    direccion: string;
    telefono: string;
    email: string;
    compras?: Venta[]; // Un cliente puede tener muchas ventas
    fechaRegistro: string;
    fechaModificacion: string;
}

export interface Venta {
    id: number;
    fecha: string;
    idCliente: number;
    cliente?: Cliente; // Relación con cliente
    total: number;
    detalles?: DetalleVenta[]; // Una venta puede tener muchos detalles
    fechaRegistro: string;
    fechaModificacion: string;
}

export interface DetalleVenta {
    id: number;
    idVenta: number;
    venta?: Venta; // Relación con venta
    idObra: number;
    obra?: Obra; // Relación con obra
}

export interface Empleado {
    id: number;
    nombre: string;
    cargo: string;
    telefono: string;
    email: string;
    fechaContratacion: string;
    fechaRegistro: string;
    fechaModificacion: string;
}